### PR TITLE
[php8-compact] Add in protection for unset variables in admin smarty …

### PR DESCRIPTION
**Pull request by seamuslee001**
…templates

Overview
----------------------------------------
This add in some if protections into various smarty templates and a couple of assignments where no variable causes problems to protect against issues in php8

Before
----------------------------------------
Hook tests fail on php8

After
----------------------------------------
Hook test doesn't fail for these classes on php8

ping @eileenmcnaughton @demeritcowboy @totten @colemanw

### CODE REVIEW

File: templates/CRM/Admin/Page/Options.tpl [hunk 1]
```diff
@@ -156,7 +156,7 @@
     </div>
 {/if}
     <div class="action-link">
-      {if $isLocked ne 1}
+      {if !empty($isLocked) and $isLocked ne 1}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:159]: I think this is wrong I think 0 would resolve to TRUE  before and FALSE after

Comment (seamuslee001) [reply templates/CRM/Admin/Page/Options.tpl]: Thanks Eileen good pick up changed it to isset now

File: templates/CRM/Admin/Page/ScheduleReminders.tpl [hunk 2]
```diff
@@ -35,7 +35,7 @@
   {/if}
   <div class="action-link">
     {assign var='link' value="civicrm/admin/scheduleReminders"}
-    {if $component}
+    {if !empty($component)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/ScheduleReminders.tpl:38]: ok

File: templates/CRM/Admin/Page/ScheduleReminders.tpl [hunk 3]
```diff
@@ -8,15 +8,15 @@
...
+{if !empty($setTab) and $setTab eq 1}
+  {if !empty($component) and $component eq 'event'}
      {include file="CRM/Event/Form/ManageEvent/Tab.tpl"}
   {/if}
 {else}
 {if $action eq 1 or $action eq 2 or $action eq 8 or $action eq 16384}
    {include file="CRM/Admin/Form/ScheduleReminders.tpl"}
 {else}
-  {if !$component}
+  {if empty($component)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/ScheduleReminders.tpl:19]: ok

File: templates/CRM/Admin/Page/ScheduleReminders.tpl [hunk 4]
```diff
@@ -8,15 +8,15 @@
  +--------------------------------------------------------------------+
 *}
 {* this template is for configuring Scheduled Reminders *}
-{if $setTab eq 1}
-  {if $component eq 'event'}
+{if !empty($setTab) and $setTab eq 1}
+  {if !empty($component) and $component eq 'event'}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/ScheduleReminders.tpl:12]: ok - we actually have been switching to && in the templates for consistency - but non blocking

File: templates/CRM/Admin/Page/RelationshipType.tpl [hunk 5]
```diff
@@ -49,10 +49,10 @@
             <td class="crm-relationship-label_b_a crm-editable" data-field="label_b_a">{$row.label_b_a}</td>
             <td class="crm-relationship-contact_type_a_display">
                 {if $row.contact_type_a_display} {$row.contact_type_a_display}
-                {if $row.contact_sub_type_a} - {$row.contact_sub_type_a} {/if}{else} {ts}All Contacts{/ts} {/if} </td>
+                {if !empty($row.contact_sub_type_a)} - {$row.contact_sub_type_a} {/if}{else} {ts}All Contacts{/ts} {/if} </td>
             <td class="crm-relationship-contact_type_b_display">
                 {if $row.contact_type_b_display} {$row.contact_type_b_display}
-                {if $row.contact_sub_type_b} - {$row.contact_sub_type_b}{/if} {else} {ts}All Contacts{/ts} {/if} </td>
+                {if !empty($row.contact_sub_type_b)} - {$row.contact_sub_type_b}{/if} {else} {ts}All Contacts{/ts} {/if} </td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/RelationshipType.tpl:55]: ok

File: templates/CRM/Admin/Page/RelationshipType.tpl [hunk 6]
```diff
@@ -49,10 +49,10 @@
             <td class="crm-relationship-label_b_a crm-editable" data-field="label_b_a">{$row.label_b_a}</td>
             <td class="crm-relationship-contact_type_a_display">
                 {if $row.contact_type_a_display} {$row.contact_type_a_display}
-                {if $row.contact_sub_type_a} - {$row.contact_sub_type_a} {/if}{else} {ts}All Contacts{/ts} {/if} </td>
+                {if !empty($row.contact_sub_type_a)} - {$row.contact_sub_type_a} {/if}{else} {ts}All Contacts{/ts} {/if} </td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/RelationshipType.tpl:52]: ok

File: templates/CRM/Admin/Form/MessageTemplates.tpl [hunk 7]
```diff
@@ -8,7 +8,9 @@
  +--------------------------------------------------------------------+
 *}
 {* this template is used for adding/editing/deleting Message Templates *}
-
+{if !isset($isAdmin)}
+  {assign var="isAdmin" value="0"}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/MessageTemplates.tpl:12]: ok

File: templates/CRM/Admin/Form/MessageTemplates.tpl [hunk 8]
```diff
@@ -50,7 +52,7 @@
         <tr>
           <td class="label-left">{$form.file_id.label}</td>
           <td>{$form.file_id.html}
-            {if $attachment}
+            {if !empty($attachment)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/MessageTemplates.tpl:55]: ok

File: templates/CRM/Admin/Form/ScheduleReminders.tpl [hunk 9]
```diff
@@ -7,6 +7,9 @@
  | and copyright information, see https://civicrm.org/licensing       |
  +--------------------------------------------------------------------+
 *}
+{if !isset($isAdmin)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/ScheduleReminders.tpl:10]: ok

File: templates/CRM/Admin/Form/ScheduleReminders.tpl [hunk 10]
```diff
@@ -85,11 +88,13 @@
         <td class="label">{$form.group_id.label}</td>
         <td>{$form.group_id.html}</td>
     </tr>
+    {if !empty($form.mode)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/ScheduleReminders.tpl:91]: ok

File: templates/CRM/Admin/Form/ScheduleReminders.tpl [hunk 11]
```diff
@@ -85,11 +88,13 @@
...
     <tr id="msgMode" class="crm-scheduleReminder-form-block-mode">
       <td class="label">{$form.mode.label}</td>
       <td>{$form.mode.html}</td>
     </tr>
-    {if $multilingual}
+    {/if}
+    {if !empty($multilingual)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/ScheduleReminders.tpl:97]: ok

File: templates/CRM/Admin/Form/ScheduleReminders.tpl [hunk 12]
```diff
@@ -124,7 +129,7 @@
        {include file="CRM/Contact/Form/Task/EmailCommon.tpl" upload=1 noAttach=1}
     </div>
     </fieldset>
-    {if $sms}
+    {if !empty($sms)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/ScheduleReminders.tpl:132]: ok

File: templates/CRM/Admin/Page/MessageTemplates.tpl [hunk 13]
```diff
@@ -125,7 +125,7 @@
                 </thead>
                 <tbody>
                 {foreach from=$template_row item=row}
-                    <tr id="message_template-{$row.id}" class="crm-entity {$row.class}{if NOT $row.is_active} disabled{/if}">
+                    <tr id="message_template-{$row.id}" class="crm-entity {if !empty($row.class)}{$row.class}{/if}{if NOT $row.is_active} disabled{/if}">
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/MessageTemplates.tpl:128]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 14]
```diff
@@ -48,7 +48,7 @@
     {else}
       {ts}You can use this page to define one or more general Email Addresses that can be selected as the From Address. EXAMPLE: <em>"Client Services" &lt;clientservices@example.org&gt;</em>{/ts}
     {/if}
-  {elseif $isLocked}
+  {elseif !empty($isLocked)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:51]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 15]
```diff
@@ -76,7 +76,7 @@
             {if !empty($hasIcons)}
               <th></th>
             {/if}
-            {if $showComponent}
+            {if !empty($showComponent)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:79]: ok

File: templates/CRM/Admin/Page/PaymentProcessorType.tpl [hunk 16]
```diff
@@ -34,7 +34,7 @@
         <tr id="paymentProcessorType-{$row.id}" class="{cycle values="odd-row,even-row"} {$row.class} crm-entity {if NOT $row.is_active} disabled{/if}">
           <td class="crm-paymentProcessorType-name">{$row.name}</td>
           <td class="crm-paymentProcessorType-title crm-editable" data-field="title">{$row.title}</td>
-            <td class="crm-paymentProcessorType-description">{$row.description}</td>
+            <td class="crm-paymentProcessorType-description">{if isset($row.description)}{$row.description}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/PaymentProcessorType.tpl:37]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 17]
```diff
@@ -58,7 +58,7 @@
 
 <div class="crm-content-block crm-block">
 {if $rows}
-{if $isLocked ne 1}
+{if !empty($isLocked) and $isLocked ne 1}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:61]: I'm still struggling with this one - should it be

if (!isset($isLocked) || $iiLocked ne 1)

ie NULL or zero should be the same - & should be TRUE

NULL would be true on the first condition & 0 on the second  so both would be TRUE

1 would be false on the first AND false on the second

Comment (seamuslee001) [reply templates/CRM/Admin/Page/Options.tpl:61]: ah I fixed it in the other place you found

Comment (seamuslee001) [reply templates/CRM/Admin/Page/Options.tpl]: Updated to be like what we have done in L159 now

File: templates/CRM/Admin/Page/Options.tpl [hunk 18]
```diff
@@ -58,7 +58,7 @@
 
 <div class="crm-content-block crm-block">
 {if $rows}
-{if $isLocked ne 1}
+{if !isset($isLocked) || $isLocked ne 1}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:61]: ok - phew

File: templates/CRM/Admin/Page/Options.tpl [hunk 19]
```diff
@@ -101,11 +101,11 @@
                 {/if}
             </th>
             {if $gName eq "payment_instrument"}<th>Account</th>{/if}
-            {if $showCounted}<th>{ts}Counted?{/ts}</th>{/if}
-            {if $showVisibility}<th>{ts}Visibility{/ts}</th>{/if}
+            {if !empty($showCounted)}<th>{ts}Counted?{/ts}</th>{/if}
+            {if !empty($showVisibility)}<th>{ts}Visibility{/ts}</th>{/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:105]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 20]
```diff
@@ -101,11 +101,11 @@
...
             <th id="nosort">{ts}Description{/ts}</th>
             <th>{ts}Order{/ts}</th>
-            {if $showIsDefault}<th>{ts}Default{/ts}</th>{/if}
+            {if !empty($showIsDefault)}<th>{ts}Default{/ts}</th>{/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:108]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 21]
```diff
@@ -117,26 +117,26 @@
             {if !empty($hasIcons)}
               <td class="crm-admin-options-icon"><i class="crm-i {$row.icon}" aria-hidden="true"></i></td>
             {/if}
-            {if $showComponent}
+            {if !empty($showComponent)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:120]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 22]
```diff
@@ -117,26 +117,26 @@
...
               <td class="crm-admin-options-component_name">{$row.component_name}</td>
             {/if}
             <td class="crm-admin-options-label crm-editable" data-field="label" {if !empty($row.color)}style="background-color: {$row.color}; color: {$row.color|colorContrast};"{/if}>
-              {$row.label}
+              {if !empty($row.label)}{$row.label}{/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:124]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 23]
```diff
@@ -117,26 +117,26 @@
...
             </td>
             {if $gName eq "case_status"}
               <td class="crm-admin-options-grouping">{$row.grouping}</td>
             {/if}
-            <td class="crm-admin-options-value">{$row.value}</td>
+            <td class="crm-admin-options-value">{if isset($row.value)}{$row.value}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:129]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 24]
```diff
@@ -117,26 +117,26 @@
...
             {if $gName eq "payment_instrument"}
               <td>{$row.financial_account}</td>
             {/if}
-            {if $showCounted}
+            {if !empty($showCounted)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:133]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 25]
```diff
@@ -117,26 +117,26 @@
...
               <td class="center crm-admin-options-filter">{icon condition=$row.filter}{ts}Counted{/ts}{/icon}</td>
             {/if}
-            {if $showVisibility}<td class="crm-admin-visibility_label">{$row.visibility_label}</td>{/if}
-            <td class="crm-admin-options-description crm-editable" data-field="description" data-type="textarea">{$row.description}</td>
-            <td class="nowrap crm-admin-options-order">{$row.weight}</td>
-            {if $showIsDefault}
+            {if !empty($showVisibility)}<td class="crm-admin-visibility_label">{$row.visibility_label}</td>{/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:136]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 26]
```diff
@@ -117,26 +117,26 @@
...
+            <td class="crm-admin-options-description crm-editable" data-field="description" data-type="textarea">{if isset($row.description)}{$row.description}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:137]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 27]
```diff
@@ -117,26 +117,26 @@
...
+            <td class="nowrap crm-admin-options-order">{if isset($row.weight)}{$row.weight}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:138]: ok

File: templates/CRM/Admin/Page/Options.tpl [hunk 28]
```diff
@@ -117,26 +117,26 @@
...
+            {if !empty($showIsDefault)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:139]: phew yes - I had to re-read a few times to understand the ifs were still intact

File: templates/CRM/Admin/Page/Options.tpl [hunk 29]
```diff
@@ -156,7 +156,7 @@
     </div>
 {/if}
     <div class="action-link">
-      {if $isLocked ne 1}
+      {if !isset($isLocked) || $isLocked ne 1}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/Options.tpl:159]: ok - NULL would be TRUE 1 would be false - this seems like where we got to

File: templates/CRM/Admin/Page/ParticipantStatusType.tpl [hunk 30]
```diff
@@ -32,7 +32,7 @@
        <tr id="participant_status_type-{$row.id}" class="crm-entity crm-participant_{$row.id} {cycle values="odd-row,even-row"} {$row.class}{if NOT $row.is_active} disabled{/if}">
           <td class="crmf-label crm-editable" data-field="label">{$row.label}</td>
           <td class="crmf-name">{$row.name} ({$row.id})</td>
-          <td class="crmf-class {if !$row.is_reserved} crm-editable {/if}" data-type="select">{$row.class}</td>
+          <td class="crmf-class {if empty($row.is_reserved)} crm-editable {/if}" data-type="select">{$row.class}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/ParticipantStatusType.tpl:35]: ok

**Comment by eileenmcnaughton** [comment]
OK - I think I've managed to confirm everything - phew!

**Pull request merged**